docs/reference/synonyms/apis/put-synonyms-set.asciidoc

```diff
@@ -7,8 +7,7 @@
 
 Creates or updates a synonyms set.
 
-NOTE: Synonyms sets are limited to a maximum of 10,000 synonym rules per set.
-Synonym sets with more than 10,000 synonym rules will provide inconsistent search results.
+NOTE: Synonyms sets are limited to a maximum of 100,000 synonym rules per set.
```

Comment:
Do we expect this increase to allowed rules to have any adverse side-effect in terms of performance?

Reply:
It will have an impact on heap when retrieving them - I need to perform some testing in order to check that we're not having too much memory usage for small clusters.

On non-API related operations, this should be no different than using file-based synonyms - and there's no limit set there.

Reply:
Should we add a note that we recommend less than 10,000?

Reply:
That makes sense - I'll be adding that.

Reply:
May be we should avoid any recommendation in terms of sizing, just put a note that for a large synonyms sets put extra demand on memory.